src/extension.ts

```diff
@@ -302,9 +292,25 @@ async function rootModules(languageClient: terraformLanguageClient, documentUri:
 	return { rootModules: rootModules, needsInit: rootModules.length === 0 };
 }
 
-async function initCommand(languageClient: terraformLanguageClient, moduleUri: string): Promise<any> {
-	const requestParams: ExecuteCommandParams = { command: `${languageClient.commandPrefix}.terraform-ls.terraform.init`, arguments: [`uri=${moduleUri}`] };
-	return execWorkspaceCommand(languageClient.client, requestParams);
+async function terraformCommand(command: string) {
+	if (vscode.window.activeTextEditor) {
+		const documentUri = vscode.window.activeTextEditor.document.uri;
+		const languageClient = getDocumentClient(documentUri);
+		const modules = await rootModules(languageClient, documentUri.toString());
+
+		let selectedModule: string;
+		if (modules.rootModules.length > 1) {
+			const selected = await vscode.window.showQuickPick(modules.rootModules.map(m => m.uri), { canPickMany: false });
+			selectedModule = selected[0];
+		} else {
+			selectedModule = modules.rootModules[0].uri;
+		}
+		const requestParams: ExecuteCommandParams = { command: `${languageClient.commandPrefix}.terraform-ls.terraform.${command}`, arguments: [`uri=${selectedModule}`] };
+		return execWorkspaceCommand(languageClient.client, requestParams);
+	} else {
+		vscode.window.showWarningMessage(`Open a module then run terraform ${command} again`);
+		return Promise.resolve();
```

Comment:
`async` keyword means you can just code like a normal func and return your value normally, it will already be wrapped in a promise.resolve. so you can just `return;`

Reply:
Good catch, thanks.